docs/src/tutorial/hello-wasm-pack.md

```diff
@@ -0,0 +1,137 @@
+# Hello, `wasm-pack`!
+
+If you open up `src/lib.rs` you should see a file that looks like this:
+
+```rust
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn it_works() {
+        assert_eq!(2 + 2, 4);
+    }
+}
+```
+
+Let's quickly modify the test suite to work for what we'll be doing. It should look like this:
+
+```rust
+#[test]
+fn it_works() {
+    assert_eq!(add(2, 2), 4);
+}
+```
+
+We'll use this later to make sure our `add` function works!
+
+Now we need to add this to the top of the file:
+
+```rust
+#![feature(use_extern_macros)]
+extern crate wasm_bindgen;
+use wasm_bindgen::prelude::*;
+```
+
+Let's step through this line by line. First up is the list of nightly features. We're enabling this for
+the whole crate. What this means is that we will later tag code with an attribute and this will
+allow Rust to generate code that we don't have to write by hand. In our case it'll use
+`wasm-bindgen.` It should be noted that `#![feature(...)]` implies using the nightly
+compiler. This gated feature will hopefully be stabilized and landed soon so that you won't need it!
+
+`wasm-bindgen` knows how to make code that works well with wasm so we don't have to
+worry about it too much and just write Rust code for the most part. If you want to know the full
+extent of its capabilities check out the README on its repo which can be found
+[here](https://github.com/alexcrichton/wasm-bindgen). For our purposes we need to know that if we
```

Comment:
Maybe

> If you want to know the full extent of its capabilities, [check out its README](https://github.com/alexcrichton/wasm-bindgen).

Reply:
Maybe also the final line could say this instead:

> For this tutorial all we need to know is that wasm-bindgen will help us generate code that lets us call our wasm code from JS without us needing to know the nitty gritty details.